apps/web/src/app/(dashboard)/documents/upload-document.tsx

```diff
@@ -96,13 +103,23 @@ export const UploadDocument = ({ className }: UploadDocumentProps) => {
     }
   };
 
+  const onFileDropRejected = () => {
+    toast({
+      title: 'File is larger than 50mb',
```

Comment:
Since this limit actually depends on the provider you use, could you create the following environment variable so users can dynamically change it for their use case?

```
# OPTIONAL: Displays the maximum document upload limit to the user in MBs
NEXT_PUBLIC_DOCUMENT_UPLOAD_LIMIT=5
```

Also could you please reword it to be `File cannot be larger than ...`

Reply:
Should i also use this env variable in the codebase in place of the limit 50mb? Or just create the env var and leave it like that?

Reply:
Yeah that'll be a good idea

Could you put it as a constant in `lib/constants/app.ts`?

Something like this but it also parses the env into number if possible

```
export const APP_DOCUMENT_UPLOAD_SIZE_LIMIT = process.env.NEXT_PUBLIC_DOCUMENT_SIZE_UPLOAD_LIMIT ?? 50;
```

Reply:
That'll make it `string | number`, we would also need to use whatever value in the `next.config.js` for the body size limits this may also apply to certain API routes.

Reply:
Yeah that's why I said it would require parsing

> we would also need to use whatever value in the next.config.js 

Not necessarily since this is purely for document upload limits

Reply:
do i need to use the document_size_limit env variable inside config in [trpc].ts files?
```
export const config = {
  maxDuration: 60,
  api: {
    bodyParser: {
      sizeLimit: '50mb',
    },
  },
};
```

Reply:
Currently the file upload transport doesn't go through TRPC, so I don't think it's relevant here

Reply:
It goes through server actions which have a similar limit configured. We also have a maxSize set on the dropzone primitive.

I think the best course of action would be to hardcode this but extract that current mix of magic numbers into a constant somewhere.